contracts/snowcones/gauge-proxy-v3.sol

```diff
@@ -0,0 +1,806 @@
+     * @param src The address of the source account
+     * @param dst The address of the destination account
+     * @param amount The number of tokens to transfer
+     * @return Whether or not the transfer succeeded
+     */
+    function transferFrom(address src, address dst, uint amount) external returns (bool) {
+        address spender = msg.sender;
+        uint spenderAllowance = allowances[src][spender];
+
+        if (spender != src && spenderAllowance != uint(-1)) {
+            uint newAllowance = spenderAllowance.sub(amount, "transferFrom: exceeds spender allowance");
+            allowances[src][spender] = newAllowance;
+
+            emit Approval(src, spender, newAllowance);
+        }
+
+        _transferTokens(src, dst, amount);
+        return true;
+    }
+
+    function _transferTokens(address src, address dst, uint amount) internal {
+        require(src != address(0), "_transferTokens: zero address");
+        require(dst != address(0), "_transferTokens: zero address");
+
+        balances[src] = balances[src].sub(amount, "_transferTokens: exceeds balance");
+        balances[dst] = balances[dst].add(amount, "_transferTokens: overflows");
+        emit Transfer(src, dst, amount);
+    }
+}
+
+contract GaugeProxyV3 is ProtocolGovernance, Strategist {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    
+    IceQueen public constant MASTER = IceQueen(0xB12531a2d758c7a8BF09f44FC88E646E1BF9D375);
+    IERC20 public constant SNOWCONE = IERC20(0x83952E7ab4aca74ca96217D6F8f7591BEaD6D64E);
+    IERC20 public constant SNOWBALL = IERC20(0xC38f41A296A4493Ff429F1238e030924A1542e50);
+    
+    IERC20 public immutable TOKEN;
+    uint public immutable DISTRIBUTION_DEADLINE = 21600;
+    
+    uint public pid;
+    uint public totalWeight;
+    uint private lockedTotalWeight;
+    uint private lockedBalance;
+    uint private locktime;
+    
+    address[] internal _tokens;
+    mapping(address => address) public gauges; // token => gauge
+    mapping(address => address) public deprecated; // token => gauge
+    mapping(address => uint) public weights; // token => weight
+    mapping(address => uint) private lockedWeights; // token => weight
+    mapping(address => mapping(address => uint)) public votes; // msg.sender => token => votes
+    mapping(address => address[]) public tokenVote;// msg.sender => token
+    mapping(address => uint) public usedWeights;  // msg.sender => total voting weight of user
+    mapping(address => bool) public deployers;
+    
+    constructor(address _governance) public {
+        TOKEN = IERC20(address(new MakeSnowCones()));
+        governance = _governance;
+    }
+
+    function addDeployer(address _deployer) external onlyGovernance {
+        deployers[_deployer] = true;
+    }
+
+    function removeDeployer(address _deployer) external onlyGovernance {
+        deployers[_deployer] = false;
+    }
+
+    modifier onlyBenevolent {
```

Comment:
For consistency, modifier could be grouped together and declared before any function.